speech-to-text/v1.ts

```diff
@@ -49,77 +50,61 @@ class SpeechToTextV1 extends GeneratedSpeechToTextV1 {
-              times: 30
-            },
-            params,
-            {
-              errorFilter: (err: SpeechToTextV1.SpeechToTextError): boolean => {
-              // if it's a timeout error, then listCorpora is called again after params.interval
-              // otherwise the error is passed back to the user
-              // if the params.times limit is reached, the error will be passed to the user regardless
-                return err.code === SpeechToTextV1.ERR_TIMEOUT;
-              }
-            }
-          );
+    try{
+      const res = await self.listCorpora(params)
+      const result = res.result;
+      if (!result.corpora.length) {
+        const sttError: SpeechToTextV1.SpeechToTextError = new Error(
+          'Customization has no corpa and therefore corpus cannot be analyzed'
+        );
+        sttError.code = SpeechToTextV1.ERR_NO_CORPORA;
+        return Promise.reject<any>(sttError)
+      }
+    } catch(err){
+      return Promise.reject<any>(err)
+    }
 
-          async.retry(
-            options,
-            (done: SpeechToTextV1.Callback<GeneratedSpeechToTextV1.Corpora>) => {
-              self.listCorpora(params, (err, res) => {
-                const corpora = res.result;
-                if (err) {
-                  done(err);
-                } else if (corpora !== undefined && isProcessing(corpora)) {
-                  // if the loop times out, async returns the last error, which will be this one.
-                  const sttError: SpeechToTextV1.SpeechToTextError = new Error(
-                    'Corpora is still being processed, try increasing interval or times params'
-                  );
-                  sttError.code = SpeechToTextV1.ERR_TIMEOUT;
-                  done(sttError);
-                } else if (corpora !== undefined && isAnalyzed(corpora)) {
-                  done(null, corpora);
-                } else {
-                  done(new Error('Unexpected corpus analysis status'));
-                }
-              });
-            },
-            next
-          );
+    const options: SpeechToTextV1.WhenCorporaAnalyzedOptions = extend(
+      {
+        interval: 5000,
+        times: 30
+      },
+      params,
+      {
+        errorFilter: (err: SpeechToTextV1.SpeechToTextError): boolean => {
+        // if it's a timeout error, then listCorpora is called again after params.interval
+        // otherwise the error is passed back to the user
+        // if the params.times limit is reached, the error will be passed to the user regardless
+          return err.code === SpeechToTextV1.ERR_TIMEOUT;
         }
-      ],
-      (err: Error | SpeechToTextV1.SpeechToTextError | null, res?: [null, GeneratedSpeechToTextV1.Corpora]) => {
-        const result = res;
-        if (err) {
-          return callback(err);
+      }
+    );
+
+    return async.retry(
+      options,
+      async (done) => {
+        try{
+          const res = await self.listCorpora(params)
```

Comment:
```suggestion
          const res = await self.listCorpora(params);
```

I'm only leaving a comment here as a reminder but a lot of statements need semicolons at the end to conform to our style guide